Firebase.agent.lib.nut

```diff
@@ -641,24 +655,53 @@ class Firebase {
                 }
                 if (200 <= res.statuscode && res.statuscode < 300) {
                     onSuccess(data);
+                    _tooManyReqTimer = false;
+                } else if (res.statuscode == 28 || res.statuscode == 429) {
+                    local now = time();
+                    // Too many requests, set _tooManyReqTimer to prevent more requests to FB
+                    if (_tooManyReqTimer == false) {
+                        // This is the first 429 we have seen set a default timeout
+                        _tooManyReqTimer = now + DEFAULT_BACK_OFF_TIMEOUT_SEC;
+                    } else if (_tooManyReqTimer <= now) {
+                        // Firebase is still overwhelmed after first timeout expired, 
+                        // Let's block requests for longer to let FB recover
+                        _tooManyReqTimer = now + (DEFAULT_BACK_OFF_TIMEOUT_SEC * _tooManyReqCounter++);
+                    }
+                    // Pass error to callback
+                    onError("Error " + res.statuscode);
                 } else if (typeof data == "table" && "error" in data) {
+                    _tooManyReqTimer = false;
                     local error = data ? data.error : null;
                     onError(error);
                 } else {
+                    _tooManyReqTimer = false;
```

Comment:
I believe we also need to handle 503 (and may be others) the same way we do for 429. But we can probably add this later if needed.

Reply:
I can add 503 to the check pretty easily.